entity-service/src/test/kotlin/com/egm/stellio/entity/web/EntityHandlerTests.kt

```diff
@@ -195,6 +195,30 @@ class EntityHandlerTests {
             .expectStatus().isBadRequest
     }
 
+    @Test
+    fun `create entity should return a 400 if input data is not valid and creation was rejected`() {
+        val jsonLdFile = ClassPathResource("/ngsild/aquac/BreedingService.json")
+
+        every { entityService.exists(any()) } returns false
+        every { entityService.createEntity(any()) } throws BadRequestDataException("Target entity does not exist")

```

Comment:
Shouldn't this be an exception with a cause to test it correctly works ?

Reply:
It aims to reproduce the behavior described in #41 and #65

Reply:
Yes indeed but the behaviour was that the Exception was wrapped inside another Exception and this was not handled (according my understanding of the issues)

Reply:
I did not want to artificially create a wrapped exception but let it be created by the framework (as in normal operations and as described in the linked issues).

But indeed I realized the exception is received already unwrapped in the exception handler in this test config ...

Reply:
> I did not want to artificially create a wrapped exception but let it be created by the framework (as in normal operations and as described in the linked issues).

This could be done in karate tests, right ?

Reply:
Yes but this is not totally satisfying, as there is still a difference of behavior at the web handler level between tests and runtime that I don't really understand ...

Anyway, in the meantime, I've modified the test to add a manual exception wrapping.